### ISSUE TITLE
[BUG] Automatically killing MSYS2 processes when updating doesn't work

### ISSUE THREAD

**Issue by HolyBlackCat** [msys2/MSYS2-packages#1991]
## Describe the bug

After updating some MSYS2 packages, instead of asking you to manually terminate all MSYS2 processes, it now asks for a confirmation do it automatically.

When I confirm, it then tells me that it can't terminate some processes, and then gives me the shell prompt.

If I try to update the packages again (without restarting the shell!), it continues the update as if nothing happened. This can't be good, because the shell was never restarted.

## Steps to Reproduce the Problem

Make a fresh install of MSYS2. Run `pacman -Syuu`.
Eventually this appears:
```
:: To complete this update all MSYS2 processes including this terminal will be closed.
    Confirm to proceed [Y/n]
```

After pressing <kbd>Enter</kbd>, I get following errors:
```
Ошибка: Не удается завершить процесс с идентификатором 6920.
Причина: Данный процесс может быть прерван только насильственно ( с  параметром /f ).
Ошибка: Не удается завершить процесс с идентификатором 6540.
Причина: Данный процесс может быть прерван только насильственно ( с  параметром /f ).
Ошибка: Не удается завершить процесс с идентификатором 5904.
Причина: Данный процесс может быть прерван только насильственно ( с  параметром /f ).
```
Then the shell prompt appears. If you run `pacman -Syuu`, it tries to continue the update (as if you restarted the terminal).

The errors mean something along the lines of: "Can't kill a process with pid X, use `/f` to force-kill it."
The pids correspond to `gpg-agent.exe`, `dirmngr.exe`, and `bash.exe`.

**Expected behavior:**
1. I assume `/f` should be passed to whatever is used to kill the processes.

and/or:

2. If the processes can't be killed, the user should probably be asked to kill the terminal session (like it was done before the update).

## Additional Context

(Not sure if any of this matters.)

- OS: Windows 7 x64
- I'm using ConEmu: `msys2_shell.cmd -conemu -mingw64 -full-path`
- ConEmu shows a little blue&yellow shield next to the tab name, so I suppose the shell is running with admin rights.

I haven't tested this without ConEmu, but can do it if needed.

**Comment by elieux** [comment]
We already call `taskkill /f`.  The elevation is probably what's interfering.  Can you check what context are pacman and the other processes running in (all elevated, all non-elevated, or mixed)?

**Comment by HolyBlackCat** [comment]
@elieux  Using the method described in [here](https://winaero.com/blog/how-to-check-if-a-process-is-running-as-administrator-elevated-in-windows/), it seems everything on this machine runs with elevated privilegies; I think UAC is disabled.

**Comment by lazka** [comment]
I've only tested with default settings an mintty, ideas on how to improve it welcome.

**Comment by elieux** [comment]
@HolyBlackCat, are you able to reproduce the situation and devise a variation of the `taskkill` command that works in your case?

**Comment by HolyBlackCat** [comment]
@elieux I tried reinstalling MSYS2, and got the same error.

But when I manually call `taskkill /PID 123 /f` from MSYS2 shell, it does work on MSYS2 processes. I had to set `MSYS2_ARG_CONV_EXCL='/PID;/f'` first though, otherwise `/f` was automatically adjusted to `F:/`, and similarly for `/PID`.

What flags are currently passed to `taskkill`? When calling it manually, I only get "use `/f` to force-kill a process" error if I omit `/f`. Are you sure `/f` is passed?
